lisa/node.py

```diff
@@ -114,6 +116,45 @@ def support_sudo(self) -> bool:
 
         return self._support_sudo
 
+    def check_sudo_password_required(self) -> None:
+        self.initialize()
+
+        # check if password is required when running command with sudo
+        if self.is_remote and self.is_posix:
+            process = self._execute("echo OK", shell=True, sudo=True, no_info_log=True)
+            result = process.wait_result(5)
+            if result.exit_code != 0 and "[sudo] password for" in result.stderr:
+                self.log.debug("Running command with sudo needs a password")
+                self.check_and_reset_password()
+
+    def check_and_reset_password(self) -> None:
+        if not self.is_remote:
+            return
+        ssh_shell = cast(SshShell, self.shell)
+        if not self.is_password_authentication_type:
+            self.log.debug(
+                "The authentication type is not password. Try to reset password..."
+            )
+            username = ssh_shell._connection_info.username
```

Comment:
use `ssh_shell.connection_info`

Reply:
SshShell doesn't have connection_info, RemoteNode has.

Reply:
ok, let's make the connection_info public.

Reply:
Done